src/server/engine_shard.cc

```diff
@@ -116,6 +118,81 @@ size_t CalculateHowManyBytesToEvictOnShard(size_t global_memory_limit, size_t gl
   return shard_budget < shard_memory_threshold ? (shard_memory_threshold - shard_budget) : 0;
 }
 
+class HuffmanCheckTask {
+ public:
+  HuffmanCheckTask() {
+    hist_.fill(0);
+  }
+
+  int32_t Run(DbSlice* db_slice);
+
+ private:
+  PrimeTable::Cursor cursor_;
+
+  static constexpr unsigned kMaxSymbol = 255;
+  array<unsigned, kMaxSymbol + 1> hist_;  // histogram of symbols.
+  string scratch_;
+};
+
+int32_t HuffmanCheckTask::Run(DbSlice* db_slice) {
+  DbTable* db_table = db_slice->GetDBTable(0);  // we currently support only default db.
+  if (!db_table)
+    return -1;
+
+  // incrementally aggregate frequency histogram.
+  auto& prime = db_table->prime;
+
+  constexpr uint32_t kMaxTraverses = 512;
```

Comment:
My concern is that you run it without flags or anything and it might freeze when iterating 512 of big strings